ParserTests/CommentsTests.cs

```diff
@@ -235,4 +461,618 @@ public void TestMixedEOLComment()
+            var token2 = tokens[1];
+            var token3 = tokens[2];
+
+            Assert.Equal(MultiLineCommentsToken.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+
+            Assert.Equal(MultiLineCommentsToken.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+
+            Assert.Equal(MultiLineCommentsToken.COMMENT, token3.TokenID);
+            Assert.Equal(@" not ending
+comment", token3.Value);
+            Assert.Equal(1, token3.Position.Line);
+            Assert.Equal(2, token3.Position.Column);
+        }
+
+        [Fact]
+        public void TestGenericMultiLineComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<MultiLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<MultiLineCommentsToken>;
+
+
+            var dump = lexer.ToString();
+
+            var code = @"1
+2 /* multi line 
+comment on 2 lines */ 3.0";
+
+            var tokens = lexer.Tokenize(code).ToList();
+
+            Assert.Equal(5, tokens.Count);
+
+            var intToken1 = tokens[0];
+            var intToken2 = tokens[1];
+            var multiLineCommentToken = tokens[2];
+            var doubleToken = tokens[3];
+
+            Assert.Equal(MultiLineCommentsToken.INT, intToken1.TokenID);
+            Assert.Equal("1", intToken1.Value);
+            Assert.Equal(0, intToken1.Position.Line);
+            Assert.Equal(0, intToken1.Position.Column);
+
+            Assert.Equal(MultiLineCommentsToken.INT, intToken2.TokenID);
+            Assert.Equal("2", intToken2.Value);
+            Assert.Equal(1, intToken2.Position.Line);
+            Assert.Equal(0, intToken2.Position.Column);
+            Assert.Equal(MultiLineCommentsToken.COMMENT, multiLineCommentToken.TokenID);
+            Assert.Equal(@" multi line 
+comment on 2 lines ", multiLineCommentToken.Value);
+            Assert.Equal(1, multiLineCommentToken.Position.Line);
+            Assert.Equal(2, multiLineCommentToken.Position.Column);
+            Assert.Equal(MultiLineCommentsToken.DOUBLE, doubleToken.TokenID);
+            Assert.Equal("3.0", doubleToken.Value);
+            Assert.Equal(2, doubleToken.Position.Line);
+            Assert.Equal(22, doubleToken.Position.Column);
+        }
+
+        [Fact]
+        public void TestGenericSingleLineComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<MultiLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<MultiLineCommentsToken>;
+
+            var dump = lexer.ToString();
+
+            var tokens = lexer.Tokenize(@"1
+2 // single line comment
+3.0").ToList();
```

Comment:
How should the parser behave here? Right now it ignores the comment as well as the DOUBLE (3.0), generating two tokens for the INT and one empty token at the end.

I would expect the comments and text to be completely ignored and the INT 2 and DOUBLE 3.0 be tokenized or to completely fail compiling with an error. The actual behavior seems erroneous for me.